build/package/Microsoft.DotNet.Cli.Nupkg.targets

```diff
@@ -0,0 +1,69 @@
+  
+  <Target Name="SetupGenerateNugetPackagesInputsOutputs"
+          DependsOnTargets="Init">
+    <PropertyGroup>
+      <NupkgVersionSuffix>$(CommitCount)</NupkgVersionSuffix>
+      <ProjectsSrcDirectory>$(RepoRoot)/src</ProjectsSrcDirectory>
+      <PackagingBuildBasePath>$(Stage2CompilationDirectory)/forPackaging</PackagingBuildBasePath>
+      <NupkgOutputDirectory>$(PackagesDirectory)</NupkgOutputDirectory>
+    </PropertyGroup>
+    
+    <ItemGroup>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.Cli.Utils" >
+        <ProjectName>Microsoft.DotNet.Cli.Utils</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-preview3-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.Compiler.Common">
+        <ProjectName>Microsoft.DotNet.Compiler.Common</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-preview3-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.Files">
+        <ProjectName>Microsoft.DotNet.Files</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-preview3-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.InternalAbstractions">
+        <ProjectName>Microsoft.DotNet.InternalAbstractions</ProjectName>
+        <Version>1.0.1-beta-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.ProjectModel">
+        <ProjectName>Microsoft.DotNet.ProjectModel</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-rc4-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.ProjectModel.Loader">
+        <ProjectName>Microsoft.DotNet.ProjectModel.Loader</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-preview3-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.DotNet.ProjectModel.Workspaces">
+        <ProjectName>Microsoft.DotNet.ProjectModel.Workspaces</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-preview3-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.Extensions.DependencyModel">
+        <ProjectName>Microsoft.Extensions.DependencyModel</ProjectName>
+        <Version>1.0.1-beta-$(CommitCount)</Version>
+      </ProjectsToPack>
+      <ProjectsToPack Include="$(ProjectsSrcDirectory)/Microsoft.Extensions.Testing.Abstractions">
+        <ProjectName>Microsoft.Extensions.Testing.Abstractions</ProjectName>
+        <Version>$(VersionMajor).$(VersionMinor).$(VersionPatch)-preview3-$(CommitCount)</Version>
```

Comment:
We should create a BuildVersion property which defaults to CommitCount. This will give us flexibility to redefine versioning scheme if ever needed.